### PR TITLE
iOS15 specific recommended zoom factor

### PR DESCRIPTION
**Pull request by ian-wd**

### CODE REVIEW

File: ios/ScannerViewController.m [hunk 1]
```diff
@@ -413,6 +418,57 @@ - (void) deinitCapture {
...
+            return;
+        }
+        
+        CMVideoDimensions formatDimensions = CMVideoFormatDescriptionGetDimensions([self.device.activeFormat formatDescription]);
+        float rectOfInterestWidth = formatDimensions.height / formatDimensions.width;
+        
+        float deviceFieldOfView = [self.device.activeFormat videoFieldOfView];
+        float minimumSubjectDistanceForCode = [self minimumSubjectDistanceForCode:deviceFieldOfView minimumCodeSize:20 previewFillPercentage:rectOfInterestWidth];
+        
+        if (minimumSubjectDistanceForCode < deviceMinimumFocusDistance) {
```
Comment (iBotPeaches) [inline ios/ScannerViewController.m:435]: Never could get this to hit in my testing.

**Review by iBotPeaches — CHANGES_REQUESTED** [review]

File: ios/ScannerViewController.m [hunk 2]
```diff
@@ -382,7 +382,12 @@ - (void)dealloc {
 
 - (void) startScanning {
     self.state = LAUNCHING_CAMERA;
-    [self.captureSession startRunning];
+    
+    dispatch_async(dispatch_get_main_queue(), ^(void) {
+        [self.captureSession startRunning];
+        [self setRecommendedZoomFactor];
+    });
```
[the remark this answers is not in the document]
Comment (ian-wd) [reply ios/ScannerViewController.m:389]: same here, it fixed when I added it last time but today it shows the same error again

File: ios/ScannerViewController.m [hunk 3]
```diff
@@ -382,7 +382,12 @@ - (void)dealloc {
 
 - (void) startScanning {
     self.state = LAUNCHING_CAMERA;
-    [self.captureSession startRunning];
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        [self.captureSession startRunning];
+        [self setRecommendedZoomFactor];
```
Comment (iBotPeaches) [inline ios/ScannerViewController.m:388]: Should this run at same interval with refocus? It seems if I understand this right it only runs on open, which I doubt the QR is perfectly placed at the right time to obtain a zoom factor.

Comment (ian-wd) [reply ios/ScannerViewController.m:388]: feels it's pretty expensive to run this along with the `reFocus` interval, maybe we should make this optional and have it integrated as a button?

Comment (iBotPeaches) [reply ios/ScannerViewController.m:388]: Ah maybe I misunderstood the video I watched. It seemed like this ran over n over to account for distance to object changing.

**Review by iBotPeaches — APPROVED** [review]

**Pull request merged**